globus_cli/services/transfer/async_delete.py

```diff
@@ -18,26 +18,24 @@
                      'asynchronous task.'))
 @common_options
 @submission_id_option
-@endpoint_id_option(help='ID of the Endpoint from which to delete file(s)')
-@click.option('--path', help='Path to the file/dir to delete')
 @click.option('--recursive', is_flag=True, help='Recursively delete dirs')
 @click.option('--ignore-missing', is_flag=True,
               help="Don't throw errors if the file or dir is absent")
 @click.option('--batch', is_flag=True,
               help=('Accept a batch of paths on stdin (i.e. run in '
-                    'batchmode). Uses --endpoint-id as passed on the '
-                    'commandline.'))
-def async_delete_command(batch, ignore_missing, recursive, path, endpoint_id,
-                         submission_id):
+                    'batchmode). Uses ENDPOINT_ID as passed on the '
+                    'commandline, and includes any commandline PATH given'))
+@click.argument('endpoint_plus_path', metavar=ENDPOINT_PLUS_OPTPATH.metavar,
+                type=ENDPOINT_PLUS_OPTPATH)
+def async_delete_command(batch, ignore_missing, recursive, submission_id,
+                         endpoint_plus_path):
     """
     Executor for `globus transfer async-delete`
     """
+    endpoint_id, path = endpoint_plus_path
     if path is None and (not batch):
```

Comment:
Just to be sure that I understand you correctly, the worry is that even with the fix (which I think is a good idea), the doomsday input results in `path=""`, right?

It seems desirable that either `path is None` or it's a valid path, coming out of the `EnpointPlusPath` types.
Do you see any problem with having the type coerce empty strings into `None`?